### ISSUE TITLE
Glossary Displays 2 or more Duplicate Sections

### ISSUE THREAD

**Issue by nrshapiro** [Psychoanalytic-Electronic-Publishing/PEP-Web-User-Interface#254]
Clicking on glossary entries is showing a significant problem in display.  It's showing multiple sections of the same definition blocks.

Examples:
<img width="608" alt="2021-01-26_12h55_28" src="https://user-images.githubusercontent.com/4837554/105904071-af219180-5fee-11eb-9312-ca5096a96380.png">

<img width="619" alt="2021-01-26_12h55_12" src="https://user-images.githubusercontent.com/4837554/105904317-fdcf2b80-5fee-11eb-8d1b-5baffebc2a94.png">

<img width="589" alt="2021-01-26_12h56_38" src="https://user-images.githubusercontent.com/4837554/105904208-da0be580-5fee-11eb-93f1-e6394da5016c.png">

In addition, some lines extend beyond the bounds of the glossary popup.

<img width="840" alt="2021-01-26_12h56_05" src="https://user-images.githubusercontent.com/4837554/105904264-eb54f200-5fee-11eb-8f9d-2e8e47347d99.png">

**Comment by nrshapiro** [comment]
@bakerac4 
Note: This seems to only happen when it's called from the Glossary Terms Infocard, rather than the link in documents.

**Comment by bakerac4** [comment]
@nrshapiro it seems like the it works that way due to not having ids. In the document text we have an id to pass usually.

In the glossary terms infocard we call 

https://stage-api.pep-web.rocks/v2/Documents/Glossary/siblingrivalry/?return_format=XML&termidtype=Group

What is the correct call to make to not get dupes?

**Comment by nrshapiro** [comment]
@bakerac4 

I've changed the way the endpoint works which should fix the problem.  The old way it worked was bringing back each term in the group as a new record, and the terms fields were supposed to be populated with the individual term data...but not all term fields were being populated.  Now, I've added a new parameter to the endpoint, to preserve that way, but the default is now to only return one record per matching group.  In my local tests, with that as the default, the client works correctly for groups that previously returned multiple entries.  The reason for the prior duplication is that despite the term fields changing (the ones that were populated anyway), the xml of the group is still the whole group.

In addition, while I was doing that, I fixed the query as well to do a more precise one.  Previously, if the glossary group entered matched a member of the group rather than the group name EXACTLY, it would fail.  So in November I had it search group_name_terms instead of group_name.  group_name_terms allows partial matches as well as exact matches, so groups with partial matches were also included.  NOW: it searches group_name first.  IF it doesn't match any groups, it expands and searches group_name_terms.  That way the terms in the glossary always match, IF they actually match a glossary term.

HOWEVER: @bakerac4 

1) You have many terms in the Glossary infocard which are MUNGEDTOGETHERASONEWORD.  These will not work.  the space between the words needs to be there.  If it's still there today (and you don't reply it's been fixed) I will bug that separately.

2) There's still the issue of the one line of some glossary entries which EXCEEDS the bounds of the popup.  Again, I will watch for a fix and bug separately.
